node/block_mining_submodule.go

```diff
@@ -0,0 +1,30 @@
+package node
+
+import (
+	"context"
+	"sync"
+
+	"github.com/filecoin-project/go-filecoin/mining"
+	"github.com/filecoin-project/go-filecoin/net/pubsub"
+	"github.com/filecoin-project/go-filecoin/protocol/block"
+)
+
+// BlockMiningSubmodule enhances the `Node` with block mining capabilities.
+type BlockMiningSubmodule struct {
+	BlockMiningAPI *block.MiningAPI
+
+	// Mining stuff.
+	AddNewlyMinedBlock newBlockFunc
+	// cancelMining cancels the context for block production and sector commitments.
+	cancelMining    context.CancelFunc
+	MiningWorker    mining.Worker
+	MiningScheduler mining.Scheduler
+	mining          struct {
+		sync.Mutex
+		isMining bool
+	}
+	miningDoneWg *sync.WaitGroup
+
+	// Network Fields
+	MessageSub pubsub.Subscription
```

Comment:
This belongs more on the messaging submodule. The block mining code does not receive these messages directly, but fetches them from the message pool (inbox).

Reply:
moved.